### PR TITLE
refactor(common): monorepo setup

### PR DESCRIPTION
**Pull request by deini**
This is a follow-up PR of https://github.com/bigcommerce/eslint-config/pull/5.

### What
- Monorepo setup
- Bump dependencies

### Why
This repo used to be a single package `@bigcommerce/eslint-config`. However, we now added a plugin. Eslint plugins are required to be their own package and be published independently.

This repo now hosts:
- `@bigcommerce/eslint-config`
- `@bigcommerce/esling-plugin`

A lerna + yarn workspaces monorepo is now in place to facilitate releases and intra-dependencies.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1,13 +1,6 @@
...
-
 module.exports = {
-  ...defaultConfig,
-  extends: [
-    // We do this filter to avoid our custom check for a prettier config on package.json
-    // for testing purposes
-    ...defaultConfig.extends.filter((config) => !config.includes('prettier')),
-    'plugin:prettier/recommended',
-  ],
+  extends: ['@bigcommerce/eslint-config'],
```
Comment (deini) [inline .eslintrc.js:2]: This the new setup we don't need the previous hack. From the workspace root we can now directly extend a config even if it comes from a workspace package.

**Review by jorgemoya — APPROVED** [review]
It looks good to me.

**Review by chanceaclark — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -1,51 +1,9 @@
...
-npm run release
-```
-
-Please refer to the documentation of [standard-version](https://github.com/conventional-changelog/standard-version) for more options.
-
-
-## License
-
-MIT
+[@bigcommerce/eslint-plugin](packages/eslint-plugin)
```
Comment (icatalina) [inline README.md:9]: Should we discourage the use of the plugin? maybe just omit it or make the main readme point to the one in `eslint-config`? 

what do u think?

File: packages/eslint-config/README.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+```
+
+If possible, try not to override the preset unless you have a special reason.
+
+## Release
+
+To release:
+
+```sh
+npm run release
```
Comment (icatalina) [inline packages/eslint-config/README.md:43]: yarn?

**Review by icatalina — APPROVED** [review]
👏

**Pull request merged**